go/vt/callerid/callerid.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2015, Google Inc. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package callerid stores/retrives CallerIDs (immediate CallerID
+// and effective CallerID) to/from the Context
+package callerid
+
+import (
+	"golang.org/x/net/context"
+)
+
+// ImmediateCallerID is the username of immediate caller to VTGate
+type ImmediateCallerID struct {
+	username string
+}
+
+// GetUsername returns the immediate caller of VTGate
+func (im *ImmediateCallerID) GetUsername() string {
+	if im == nil {
+		return ""
+	}
+	return im.username
+}
+
+// NewImmediateCallerID creates a ImmediateCallerID initialized with username
+func NewImmediateCallerID(username string) *ImmediateCallerID {
+	return &ImmediateCallerID{username: username}
+}
+
+// EffectiveCallerID is the identity of the actual caller to Vitess, for example,
+// if the actual caller sends requests to Vitess through a proxy, then the EffectiveCallerID
+// will contain information about the actual caller while the ImmediatieCallerID has
+// the username running the proxy
+type EffectiveCallerID struct {
+	principal    string
+	component    string
+	subComponent string
+}
```

Comment:
For this one, just use the proto3 vtrpc.CallerID structure. Again, no need to re-invent another structure.